src/libsystemd-network/dhcp-server-internal.h

```diff
@@ -48,6 +50,7 @@ struct sd_dhcp_server {
         int fd_raw;
 
         int ifindex;
+        char ifname[IFNAMSIZ + 1];
```

Comment:
i'd just make this a `char*`, and store the name in dynamic memory with any length, so that people can pass altnames too